pkg/commands/status_discovery.go

```diff
@@ -53,19 +61,51 @@ func (s *CloudDiscoveryStatusStore) GetApiIngressStatus(cluster *kops.Cluster) (
 
 	if awsCloud, ok := cloud.(awsup.AWSCloud); ok {
 		name := "api." + cluster.Name
-		lb, err := awstasks.FindLoadBalancerByNameTag(awsCloud, name)
-		if lb == nil {
-			return nil, nil
+
+		ELB, NLB := false, false
+
+		lbSpec := cluster.Spec.API.LoadBalancer
+		switch lbSpec.Class {
+		case kops.LoadBalancerClassClassic, "":
+			ELB = true
+		case kops.LoadBalancerClassNetwork:
+			NLB = true
+		default:
+			return nil, fmt.Errorf("Unknown Cluster.Spec.API.LoadBalancer.Class : %v", lbSpec.Class)
```

Comment:
I think we should add an api validation in `pkg/apis/kops/validation/validation.go` that ensures the class is one of the supported values. Then this lower-level code doesn't have to worry about unsupported values.